src/errno.rs

```diff
@@ -1,24 +1,31 @@
+#![macro_use]
+
 extern crate libc;
+
 use std::ffi::CString;
+use std::ffi::CStr;
 use std::os::raw::c_char;
 use libc::c_int;
 use fileops::open;
 
-pub fn strerrer(errno: c_int) -> String {
+pub fn strerror(errno: c_int) -> String {
   unsafe {
-    let c_err_str = libc::strerror(errno);
-    CString::from_raw(c_err_str).into_string()
-      .unwrap_or(String::from("Unknown errno"))
+    CStr::from_ptr(libc::strerror(errno)).to_string_lossy().into_owned()
   }
 }
 
 pub fn errno() -> c_int {
   unsafe { *libc::__errno_location() }
 }
 
+macro_rules! dump_errno_str {
+  () => (format!("errno: {}, errno str {}", errno(), strerror(errno())))
```

Comment:
So yeah, the macro will work if you import the things its trying to use. If macros were truly hygienic it wouldn't though I think. For portability I think its best to import locally at invocation site.